File: App/app/src/main/java/com/cinema/ticket_booking/ui/search/SearchViewModel.java
package com.cinema.ticket_booking.ui.search;

import androidx.lifecycle.*;
import com.cinema.ticket_booking.data.model.response.*;
import com.cinema.ticket_booking.data.repository.CinemaRepository;
import com.cinema.ticket_booking.util.Resource;
import dagger.hilt.android.lifecycle.HiltViewModel;
import javax.inject.Inject;
import java.util.List;

@HiltViewModel
public class SearchViewModel extends ViewModel {
    private final CinemaRepository repo;
    private final MutableLiveData<String> cityTrigger = new MutableLiveData<>();
    private final LiveData<Resource<List<CinemaResponse>>> cinemas;

    @Inject
    public SearchViewModel(CinemaRepository repo) {
        this.repo = repo;
        this.cinemas = Transformations.switchMap(cityTrigger, repo::getCinemas);
    }

    public LiveData<Resource<List<CinemaResponse>>> getCinemas() {
        return cinemas;
    }

    public void loadCinemas(String city) {
        cityTrigger.setValue(city);
    }
}

File: App/app/src/main/java/com/cinema/ticket_booking/ui/search/SearchViewModel.kt
package com.cinema.ticket_booking.ui.search

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.switchMap
import com.cinema.ticket_booking.data.model.response.CinemaResponse
import com.cinema.ticket_booking.data.repository.CinemaRepository
import com.cinema.ticket_booking.util.Resource
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

@HiltViewModel
class SearchViewModel @Inject constructor(
    private val repo: CinemaRepository
) : ViewModel() {
    private val cityTrigger = MutableLiveData<String>()
    val cinemas: LiveData<Resource<List<CinemaResponse>>> = cityTrigger.switchMap { city ->
        repo.getCinemas(city)
    }

    fun loadCinemas(city: String) {
        cityTrigger.value = city
    }
}
